util/fibers/uring_proactor.cc

```diff
@@ -310,12 +314,28 @@ SubmitEntry UringProactor::GetSubmitEntry(CbType cb, uint16_t submit_tag) {
   return SubmitEntry{res};
 }
 
-int UringProactor::RegisterBuffers(const struct iovec* iovecs, unsigned nr_vecs) {
-  return io_uring_register_buffers(&ring_, iovecs, nr_vecs);
+int UringProactor::RegisterBuffers(size_t size) {
+  size = (size + 4096 - 1) / 4096 * 4096;
+  buf_pool_.backing = new (std::align_val_t(4096)) uint8_t[size];
```

Comment:
i think it's better to use `mmap`  instead of going through our allocator layer.
mi_malloc, for example, allocates `size + alignment` to guarantee alignment, and frankly calling `new` here is redundant given we want 4kb multiple and alignment.